### PR TITLE
feature: introduce the fault-injection plugin.

### LINKED ISSUE #1038: feature: I want to introduce a fault-injection plugin
Hi @moonming @membphis 

I want to introduce a plugin fault-injection like envoy's [fault-filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter).

The purpose of introducing:
> The fault injection filter can be used to test the resiliency of microservices to different forms of failures. The filter can be used to inject delays and abort requests with user-specified error codes, thereby providing the ability to stage different failure scenarios such as service failures, service overloads, high network latency, network partitions, etc. 

In the first phase, I implement a very simple plugin which simply supports inject delay and abort requests with user-specified error codes. 

What do you think of?
Thanks.

### PR DESCRIPTION
**Pull request by agile6v**
### Issues resolved

Fix #1038

### CODE REVIEW

File: lua/apisix/plugins/fault-injection.lua [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                http_status = {type = "integer", minimum = 200},
+                body = {type = "string", minLength = 0},
+            },
+            required = {"http_status"}
+        },
+        delay = {
+            type = "object",
+            properties = {
+                duration = {
+                    type = "string",
```
Comment (membphis) [inline lua/apisix/plugins/fault-injection.lua]: it should be a `number`

Comment (membphis) [reply lua/apisix/plugins/fault-injection.lua]: If the `delay` object only has one `property`, I think the type of `delay` should be "number".

`delay = {type = "number", maximum = 100, minimum = 0}`

File: t/plugin/fault-injection.t [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+                               "fault-injection": {
+                                   "delay": {
+                                       "duration": "1"
+                                   }
+                               },
+                               "proxy-rewrite": {
+                                   "uri": "/hello"
+                               }
+                           },
+                           "upstream": {
```
Comment (membphis) [inline t/plugin/fault-injection.t]: we can remove `upstream` object, it is useless. please fix some other similar points.

**Comment by membphis** [comment]
@agile6v missing a new doc about how to use this plugin.

**Comment by agile6v** [comment]
> missing doc about how to use this plugin. we can submit it in a separate new PR.

OK. I will submit the documentation later.

**Comment by membphis** [comment]
@agile6v merged, many thx

**Pull request merged**